lib/phoenix/tracker/state.ex

```diff
@@ -83,6 +83,27 @@ defmodule Phoenix.Tracker.State do
     add(state, pid, topic, key, meta)
   end
 
+  @doc """
+  Updates an element via leave and join.
+
+  Atomically updates ETS local entry.
+  """
+  @spec leave_join(t, pid, topic, key, meta) :: t
+  def leave_join(%State{delta: delta} = state, pid, topic, key, meta) do
+    # Produce remove-like delta
+    [{{^topic, ^pid, ^key}, _meta, tag}] = :ets.lookup(state.values, {topic, pid, key})
+    pruned_clouds = delete_tag(state.clouds, tag)
+    new_delta = remove_delta_tag(state.delta, tag)
+    state = bump_clock(%State{state | clouds: pruned_clouds, delta: new_delta})
+
+    # Update ETS entry and produce add-like delta
+    state = bump_clock(state)
```

Comment:
Not sure if we need to bump the clock twice, I just mirrored the exact behaviour of leave + join.